src/openforms/js/components/formio_builder/translation.js

```diff
@@ -241,7 +242,9 @@ export const handleComponentValueLiterals = (
       return null;
   }
   let translations = {};
-  values.forEach(({label}, index) => {
+  const nonEmptyValues = values.filter(item => !isEmpty(item));
+  if (!nonEmptyValues.length) return null;
+  nonEmptyValues.forEach(({label = ''}, index) => {
```

Comment:
this is the crucial fix for the infinite loops - looks like formio for its data values has a default value of `[{}`], so the `label` inside the object is undefined. It uses that even if we have the data source set to variable rather than literal options.

It looks like the `undefined` label then caused an infinite cycle of change triggers. Filtering out these empty objects (there's no literal to translate anyway) stopped the infinite loops.